src/Chapter4.hs

```diff
@@ -629,7 +651,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM b1 b2 = b1 >>= (\b -> if not b then pure b else pure (b &&) <*> b2)
```

Comment:
Yes, your implementation is a bit overcomplicated, but don't worry! It is just a matter
```suggestion
andM b1 b2 = b1 >>= (\b -> if b then b2 else pure False)
```